src/inject.ts

```diff
@@ -363,6 +380,8 @@ async function injectionAlreadyExists({
 	summary.INCOMPLETE_CANDIDATES += isComplete ? 0 : 1;
 	if (isComplete) {
 		await deleteTorrentFileIfSafe(torrentFilePath);
+	} else {
+		deleteTorrentFileIfComplete(torrentFilePath, meta.infoHash);
```

Comment:
This one looks redundant?

Reply:
`isComplete` is manually set to true in a scenario above to prevent infinite rechecks.

Reply:
Would it be possible to just delete directly there rather than setting iscomplete?

Reply:
We make a call to client for completion at the start which most likely will be true. It gets overriden if it was false but all files are present (corrupted torrent). Doing it this way prevents an unnecessary call most times this function is ran.

Reply:
Which call is unnecessary?

Reply:
We need to check if a torrent is complete for already exists logic. If it is, we can use that value to delete instead of the unnecessary request the wait function will make. That with the fact that it can be overriden, means it's best handle the checks like this.